packages/tools/src/types/ContourTypes.ts

```diff
@@ -30,5 +34,8 @@ export type ContourRenderingConfig = {
 export type ContourSegmentationData = {
   // Ids of the contourSets that are part of this segmentation
   // in the cache
-  geometryIds: string[];
+  geometryIds?: string[];
+  // Ids of the annotations that are part of this segmentation
+  // grouped by segmentIndex
+  annotationUIDsMap?: Map<number, Set<string>>;
 };
```

Comment:
Please understand that it doesn't make sense to have both options. In the long term, we are moving towards the annotationUIDsMap.